main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -36,21 +36,48 @@ object GenExpression {
 
   type Ref[T] = Array[T]
 
-  case class MethodContext(clazz: JvmType.Reference,
+  sealed trait MethodContext {
+
+    def clazz: JvmType.Reference
+
+    def entryPoint: Label
+
+    def lenv: Map[Symbol.LabelSym, Label]
+
+    def localOffset: Int
+
+    def addLabels(labels: Map[Symbol.LabelSym, Label]): MethodContext = {
+      val updatedLabels = this.lenv ++ labels
+      this match {
+        case ctx: EffectContext =>
+          ctx.copy(lenv = updatedLabels)
+        case ctx: DirectContext =>
+          ctx.copy(lenv = updatedLabels)
+      }
+    }
+
+  }
+
+  case class EffectContext(clazz: JvmType.Reference,
                            entryPoint: Label,
                            lenv: Map[Symbol.LabelSym, Label],
                            newFrame: InstructionSet, // [...] -> [..., frame]
                            setPc: InstructionSet, // [..., frame, pc] -> [...]
```

Comment:
Oh i forgot this part. You can see that setPc expects frame and PC on the stack, so we have to avoid those in the direct case